packages/kbn-ui-shared-deps-src/src/polyfills.js

```diff
@@ -17,3 +17,67 @@ if (typeof window.Event === 'object') {
 
 require('whatwg-fetch');
 require('symbol-observable');
+
+/* @notice
+ *
+ * Adapted from remote-web-worker, which was available under a "MIT" license.
+ *
+ * MIT License (MIT)
+ *
+ * Copyright (c) 2022 Jan Nicklas
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+typeof window !== 'undefined' &&
+  // eslint-disable-next-line no-global-assign
+  (Worker = ((BaseWorker) =>
+    class Worker extends BaseWorker {
+      constructor(scriptURL, options) {
+        const url = String(scriptURL);
+        super(
+          // Check if the URL is remote
+          url.includes('://') && !url.startsWith(window.location.origin) && !url.startsWith('blob:') // to bootstrap the actual script to work around the same origin policy.
+            ? URL.createObjectURL(
```

Comment:
Observation/nit: Im not sure how severe is, but I think this may have a [small memory leak](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL_static) - seems low impact, just something to keep in mind!